flambe/metric/dev/binary.py

```diff
@@ -244,3 +244,35 @@ def __str__(self) -> str:
         """Return the name of the Metric (for use in logging)."""
         invert_label = "Negative" if self.positive_label == 0 else "Positive"
         return f"{invert_label}{self.__class__.__name__}"
+
+
+class F1(BinaryMetric):
+
+    def __init__(self, threshold: float = 0.5, positive_label: int = 1):
+        super().__init__(threshold)
+        self.recall = BinaryRecall(threshold, positive_label)
+        self.precision = BinaryPrecision(threshold, positive_label)
+
+    def compute_binary(self,
+                       pred: torch.Tensor,
+                       target: torch.Tensor) -> torch.Tensor:
+        """Compute F1. Score, the harmonic mean between precision and
+        recall.
+        Parameters
+        ---------
+        pred: torch.Tensor
+            Predictions made by the model. It should be a probability
+            0 <= p <= 1 for each sample, 1 being the positive class.
+        target: torch.Tensor
+            Ground truth. Each label should be either 0 or 1.
+
+        Returns
+        ------
+        torch.float
+            The computed binary metric
+
+        """
+        recall = self.recall.compute_binary(pred, target)
+        precision = self.precision.compute_binary(pred, target)
+
+        return 2 * precision * recall / (precision + recall + 1e-10)
```

Comment:
Nice implementation reusing what we have. Is this the best way to handle `0` division?

Reply:
I suggest using an `eps` argument to the object with a default of anything between 1e-6 and 1e-10, and use `self.eps` here so it's controllable